File: benchmarks/src/main/resources/python/sort_intarray_starter.py
import sys
from random import randint
from sort_intarray import sort

def swap(a, i, j):
  tmp = a[i]
  a[i] = a[j]
  a[j] = tmp

def create_shuffled_array(n):
  array = []
  for index in range(0, n):
    array.append(index)

  for index in reversed(range(1, n)):
    swap_with = randint(0, index)
    swap(array, index, swap_with)

  return array

n = int(sys.argv[1])

arr = create_shuffled_array(n)
sort(arr)
print(str(arr))

File: benchmarks/src/main/resources/js/sort.intarray.starter.js
function createShuffledArray(n) {
  let array = new Array(n)
  for (let index = 0; index < n; ++index) {
    array[index] = index;
  }
  for (let index = n - 1; index >= 0; --index) {
    let min = 0;
    let max = n;
    let swapWith =  Math.floor(Math.random() * (max - min + 1)) + min;
    swap(array, index, swapWith)
  }
  return array
}

function swap(array, i, j) {
  let tmp = array[i]
  array[i] = array[j]
  array[j] = tmp
}

for (let run = 1; run <= 10; ++run) {
  let shuffled = createShuffledArray(8388608)

  console.time('sort')
  shuffled.sort()
  console.timeEnd('sort')
}